crates/metaslang/cst/src/cst.rs

```diff
@@ -62,21 +90,43 @@ impl<T: KindTypes> Node<T> {
         Self::Token(Rc::new(TerminalNode { kind, text }))
     }
 
+    pub fn id(&self) -> usize {
+        match self {
+            Self::Rule(node) => Rc::as_ptr(node) as usize,
+            Self::Token(node) => Rc::as_ptr(node) as usize,
+        }
+    }
+
+    pub fn kind(&self) -> NodeKind<T> {
+        match self {
+            Self::Rule(node) => NodeKind::NonTerminal(node.kind),
+            Self::Token(node) => NodeKind::Terminal(node.kind),
+        }
+    }
+
     pub fn text_len(&self) -> TextIndex {
         match self {
             Self::Rule(node) => node.text_len,
             Self::Token(node) => (&node.text).into(),
         }
     }
 
-    /// Returns a slice of the children (not all descendants) of this node.
-    pub fn children(&self) -> &[LabeledNode<T>] {
+    pub fn edges(&self) -> &[Edge<T>] {
         match self {
             Self::Rule(node) => &node.children,
             Self::Token(_) => &[],
         }
     }
 
+    pub fn labeled_edges(&self) -> impl Iterator<Item = &Edge<T>> {
+        self.edges().iter().filter(|edge| edge.label.is_some())
+    }
+
+    /// Returns a slice of the children (not all descendants) of this node.
+    pub fn children(&self) -> &[Edge<T>] {
```

Comment:
style/general nit: Rather than having two functions with different names on the same concrete type, I think it'd be cleaner to:
- keep the more obvious or related more immediately to the concrete type terms like `children` (since this is parent/child 'contains' relationship rather than a general graph that happens to be a tree)
- provide a separate, more abstract trait, which this concrete type would implement, that might provide a more abstract functionality like `edges`.

This way, one could use the higher-level abstraction using the traits (I imagine we will want to define that for subsequent tree/node-traversable types) if they need but if they're using a concrete type, they will use the immediate/more appropriate name.

Reply:
We should probably also expose `labeled_children` as well, if we aim to keep the symmetry here.

Reply:
It should be `edge` in any case ...

Reply:
This feels like a drive-by. We have two other issues looking at this (renaming, api review)

Reply:
I suggest adding this to the review issue, so that we don't forget about it.